### DISCUSSION TITLE
Modyfying recipe_radiation_budget.yml

### DISCUSSION THREAD

**Discussion by AnushaNTNU** [ESMValGroup/ESMValCore#1745] [Q&A] [answered]
Hi
I just changed the models in the recipe recipe_radiation_budget.yml and i got the error

2022-10-05 14:03:45,350 UTC [260341] ERROR   Program terminated abnormally, see stack trace below for more information:
Traceback (most recent call last):
  File "/conda/esmvaltool/2.6.0/lib/python3.10/site-packages/esmvalcore/_main.py", line 499, in run
    fire.Fire(ESMValTool())
  File "/conda/esmvaltool/2.6.0/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/conda/esmvaltool/2.6.0/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/conda/esmvaltool/2.6.0/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/conda/esmvaltool/2.6.0/lib/python3.10/site-packages/esmvalcore/_main.py", line 443, in run
    process_recipe(recipe_file=recipe, config_user=cfg)
  File "/conda/esmvaltool/2.6.0/lib/python3.10/site-packages/esmvalcore/_main.py", line 121, in process_recipe
    recipe = read_recipe_file(recipe_file, config_user)
  File "/conda/esmvaltool/2.6.0/lib/python3.10/site-packages/esmvalcore/_recipe.py", line 70, in read_recipe_file
    check.recipe_with_schema(filename)
  File "/conda/esmvaltool/2.6.0/lib/python3.10/site-packages/esmvalcore/_recipe_checks.py", line 50, in recipe_with_schema
    yamale.validate(schema, recipe, strict=False)
  File "/conda/esmvaltool/2.6.0/lib/python3.10/site-packages/yamale/yamale.py", line 43, in validate
    raise YamaleError(results)
yamale.yamale_error.YamaleError: Error validating data '/projects/NS9576K/anusha/ESMValTool/esmvaltool/recipes/recipe_radiation_budget_AS.yml' with schema '/conda/esmvaltool/2.6.0/lib/python3.10/site-packages/esmvalcore/recipe_schema.yml'
        datasets.0.dataset: Required field missing
2022-10-05 14:03:45,399 UTC [260341] INFO
If you have a question or need help, please start a new discussion on https://github.com/ESMValGroup/ESMValTool/discussions
If you suspect this is a bug, please open an issue on https://github.com/ESMValGroup/ESMValTool/issues
To make it easier to find out what the problem is, please consider attaching the files run/recipe_*.yml and run/main_log_debug.txt from the output directory

Does anyone know what is this?

Thanks

**Comment by remi-kazeroni** [comment, marked as answer]
Hi @AnushaNTNU, thanks for reporting the issue. To get a better understanding of the issue encountered, could you please post here the `recipe_radiation_budget_AS.yml` and `main_log_debug.txt` files? These can be found in your output directory, e.g. `~/esmvaltool_output/recipe_radiation_budget_AS_YYYYMMDD_HHMMSS/run/`. Note that you will need to rename the recipe file with a `.txt` at the end since yml files can't be posted directly on GitHub. If needed, you can have a look at [this page](https://docs.github.com/en/github-ae@latest/get-started/writing-on-github/working-with-advanced-formatting/attaching-files) on how to post files on github.

**Reply by AnushaNTNU** [reply]
Thanks for the reply. I am attaching the files herewith
[main_log_debug.txt](https://github.com/ESMValGroup/ESMValCore/files/9723043/main_log_debug.txt)
[recipe_radiation_budget_AS.txt](https://github.com/ESMValGroup/ESMValCore/files/9723044/recipe_radiation_budget_AS.txt)

**Reply by remi-kazeroni** [reply]
Thanks for sharing the files @AnushaNTNU. The issue you're encountering is due to a missing space after ":" in the definition of the first dataset of your recipe line 21. That line should be:
```yml
  - {dataset: CanESM5, project: CMIP6, exp: ssp534-over-bgc,
```
Please let me know if that works for you. Your recipe seems to run fine for me.

Also, it is recommended to have one only one space after each ":" for key-value pairs in the recipe file. For example: `exp: ssp534-over-bgc`.

**Reply by AnushaNTNU** [reply]
Hi @remi-kazeroni 
Thanks a lot. I didnt know its case sensitive.  It is succesful now. I have another question about the recipe. Since my dataset of interest is future simualtions (SSP5.3.4OS), I cannot compare my output with obs right though i got some nice plost in output which is wrong. So can I remove that part from recipe? I just want to compare radiation budget of all models that simulate this particular scenario.

**Reply by remi-kazeroni** [reply]
You're welcome @AnushaNTNU. I'm not so sure obs datasets could be easily commented out of recipes. This would depend on how the diagnostic scripts used in this recipe work. But I can't really advise here since I didn't contribute to this recipe. I would recommend to ask the authors of the recipe: @Jon-Lillis and @ehogan.

**Reply by AnushaNTNU** [reply]
Ok Thanka a lot. I will check the diagnostic script and try to contact them. I will close this case as it is already answered.
Thanks again
